### DISCUSSION TITLE
[v5] realations with restrictions?  possible?

### DISCUSSION THREAD

**Discussion by rbartholomay** [adonisjs/core#1526] [Help] [answered]
Hi!

is it possible to define relations with restrictions in v5 inside the model?

```javascript
  @belongsTo(() => CollectorIdrCatalog, {
    localKey: 'item_id',
    foreignKey: 'parent_item_id',
   // where objectType = 6 ???   
  })
  public contactcenter: BelongsTo<typeof CollectorIdrCatalog>
```

Is this possible?

Bye René

**Comment by thetutlage** [comment, marked as answer]
Covered in docs already https://preview.adonisjs.com/guides/model-relations/introduction/#on-query-hook

**Reply by rbartholomay** [reply]
unbelievable...    This is really great!!!
